src/ray/object_manager/common.cc

```diff
@@ -74,10 +74,33 @@ Status PlasmaObjectHeader::CheckHasError() const {
 
 #if defined(__APPLE__) || defined(__linux__)
 
-Status PlasmaObjectHeader::TryToAcquireSemaphore(sem_t *sem) const {
+Status PlasmaObjectHeader::TryToAcquireSemaphore(
+    sem_t *sem,
+    const std::unique_ptr<std::chrono::steady_clock::time_point> &timeout_point) const {
   // Check `has_error` first to avoid blocking forever on the semaphore.
   RAY_RETURN_NOT_OK(CheckHasError());
-  RAY_CHECK_EQ(sem_wait(sem), 0);
+  auto now = std::chrono::steady_clock::now();
+
+  if (!timeout_point) {
+    RAY_CHECK_EQ(sem_wait(sem), 0);
+  } else if (now >= *timeout_point) {
```

Comment:
This `else if` seems to be redundant. The logic in `else` seems to be enough.

Reply:
Yeah we can refactor to use do..while. See Jack's comment.